src/generic_factory.h

```diff
@@ -304,7 +304,7 @@ class generic_factory
                 }
 
             } else if( jo.has_array( legacy_id_member_name ) ) {
-                for( const auto &e : jo.get_array( legacy_id_member_name ) ) {
+                for( const JsonValue &e : jo.get_array( legacy_id_member_name ) ) {
```

Comment:
```suggestion
                for( const JsonValue e : jo.get_array( legacy_id_member_name ) ) {
```
```
In file included from src/achievement.cpp:15:
src/generic_factory.h:307:39: error: loop variable 'e' is always a copy because the range of type
      'JsonArray' does not return a reference [-Werror,-Wrange-loop-analysis]
                for( const JsonValue &e : jo.get_array( legacy_id_member_name ) ) {
                                      ^
src/achievement.cpp:404:25: note: in instantiation of member function
      'generic_factory<achievement>::load' requested here
    achievement_factory.load( jo, src );
                        ^
src/generic_factory.h:307:22: note: use non-reference type 'JsonValue'
                for( const JsonValue &e : jo.get_array( legacy_id_member_name ) ) {
                     ^~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [obj/tiles/achievement.o] Error 1
```